packages/dds/test-dds-utils/src/ddsFuzzHarness.ts

```diff
@@ -662,17 +682,43 @@ export function mixinAttach<
 			// However, now that we're transitioning to an attached state, the summarizer client should never have any edits.
 			// Thus we use one of the clients we just loaded as the summarizer client, and keep the client around that we generated the
 			// attach summary from.
-			const summarizerClient = clients[0];
-			clients[0] = state.clients[0];
+			let summarizerClient = clients[0];
+			if (clientA.channel.id === "summarizer") {
+				summarizerClient = clientA;
+			} else {
+				clients[0] = state.clients[0];
+			}
 
 			return {
 				...state,
 				isDetached: false,
 				clients,
 				summarizerClient,
 			};
-		}
+		} else if (operation.type === "rehydrate") {
+			const clientA = state.clients[0];
+			assert.equal(state.clients.length, 1);
+			state.isDetached = true;
 
+			const containerRuntimeFactory = new MockContainerRuntimeFactoryForReconnection(
+				options.containerRuntimeOptions,
+			);
+			const summarizerClient = await loadDetached(
+				containerRuntimeFactory,
+				clientA,
+				model.factory,
+				"summarizer",
```

Comment:
this should match how createDetachedClient is used, so i believe should always be  `makeFriendlyClientId(random, 0)`

i think this will let you remove the change above. basically, the summarzier client should never have local changes applied, so it shouldn't be the detached client

Reply:
there is a comment above that also explains this